lib/validation/validator.js

```diff
@@ -1,37 +1,33 @@
 import {fileURLToPath} from "node:url";
 import {readFile} from "node:fs/promises";
 
-async function loadSchema(schemaPath) {
-	const filePath = schemaPath.replace("http://ui5.sap/schema/", "");
-	const schemaFile = await readFile(
-		fileURLToPath(new URL(`./schema/${filePath}`, import.meta.url)), {encoding: "utf8"}
-	);
-	return JSON.parse(schemaFile);
-}
-
 class Validator {
 	constructor({Ajv, ajvErrors}) {
 		this.ajv = new Ajv({
 			allErrors: true,
 			jsonPointers: true,
-			loadSchema
+			loadSchema: Validator.loadSchema
 		});
 		ajvErrors(this.ajv);
 	}
 
-	_compileSchema() {
+	_compileSchema(schemaName) {
```

Comment:
The current implementation of _compileSchema doesn't support multiple schemas as it caches the compiled fn in a promise. I've added a failing test to demonstrate this.

I think the easiest solution would be to just have two independent instances of `Validator`, one for each schema. This way it's ensure that they don't interfere with each other. My new test would then become obsolete.

Reply:
How about now?
It's been limited to the schemas that we support, but still uses (almost) the same interface.
The caching issues that we had, now should be addressed and not face them anymore. Even if we try to extend the validator with some other schema, now should be easier- just extend the enum with the available schemas

Reply:
Issues look resolved 👍🏻
However I now feel like, as the two approaches have been combined, it is a bit over-engineered.

When we use separate instances for each schema, why don't we pass the schema name to the constructor so that this instance always and only validates that specific schema?

On the other hand, when adjusting _compileSchema to work with multiple schemas I don't see a reason to have multiple instances. But I might have missed some detail here.

Reply:
The shcemaName as an argument to the constructor looks like the correct approach. In that case, we'll avoid ambiguities about which schema is actually being validated. I'll refactor it. Thanks!